### PR TITLE
Update node backend main service

### PR DESCRIPTION
**Pull request by cazapatamar**

### CODE REVIEW

File: services/backend/src/routes/protected_areas.js [hunk 1]
```diff
@@ -22,19 +22,16 @@ module.exports = (errorHandler, paService) => {
...
-      }
-      return paService.getPAAreas(req.params.areaType, req.params.areaId).then((categories) => {
-        res.send(categories);
-        next();
-      });
-    }),
-  );
+  router.get('/pa', (req, res, next) => {
+    if (!(req.params.areaType && req.params.areaId)) {
+      const error = { code: 400, message: 'areaType and areaId required' };
```
Comment (erikasv) [inline services/backend/src/routes/protected_areas.js:27]: change this error

File: services/backend/src/routes/protected_areas.js [hunk 2]
```diff
@@ -57,21 +54,18 @@ module.exports = (errorHandler, paService) => {
...
-        .getPAAreas(req.params.areaType, req.params.areaId, req.params.seType)
-        .then((binaryProtected) => {
-          res.send(binaryProtected);
-          next();
-        });
-    }),
-  );
+  router.get('/pa/se', (req, res, next) => {
+    if (!(req.params.areaType && req.params.areaId && req.params.seType)) {
+      const error = { code: 400, message: 'areaType, areaId and seType required' };
```
Comment (erikasv) [inline services/backend/src/routes/protected_areas.js:59]: change this error

File: services/backend/src/service/project.js [hunk 3]
```diff
@@ -37,7 +38,7 @@ module.exports = (projectPersistence, biomeService) => ({
 
     const noProjectProp = groupProps.filter((prop) => !(prop in projects[0]));
     if (noProjectProp.length !== 0) {
-      throw new ReferenceError(`Some of ${groupProps} are not project properties`);
+      throw new RestifyErrors.NotFoundError(`Some of ${groupProps} are not project properties`);
```
Comment (erikasv) [inline services/backend/src/service/project.js:41]: this should be a bad request

File: services/backend/src/service/project.js [hunk 4]
```diff
@@ -53,9 +54,7 @@ module.exports = (projectPersistence, biomeService) => ({
   getProjectById: async (projectId) => {
     const pId = parseInt(projectId, 10);
     if (!pId) {
-      const error = new Error('Missing or invalid project id');
-      error.code = 400;
-      throw error;
+      throw new RestifyErrors.NotFoundError('Missing or invalid project id');
```
Comment (erikasv) [inline services/backend/src/service/project.js:57]: bad request too

File: services/backend/src/service/project.js [hunk 5]
```diff
@@ -85,9 +84,7 @@ module.exports = (projectPersistence, biomeService) => ({
   addBiomes: async (projectId, biomes) => {
     const pId = parseInt(projectId, 10);
     if (!pId) {
-      const error = new Error('Invalid project id');
-      error.code = 400;
-      throw error;
+      throw new RestifyErrors.NotFoundError('Invalid project id');
```
Comment (erikasv) [inline services/backend/src/service/project.js:87]: bad request

File: services/backend/src/service/project.js [hunk 6]
```diff
@@ -111,9 +108,7 @@ module.exports = (projectPersistence, biomeService) => ({
   getDecisionTree: async (projectId) => {
     const pId = parseInt(projectId, 10);
     if (!pId) {
-      const error = new Error('Invalid project id');
-      error.code = 400;
-      throw error;
+      throw new RestifyErrors.NotFoundError('Invalid project id');
```
Comment (erikasv) [inline services/backend/src/service/project.js:111]: bad request

File: services/backend/src/service/project.js [hunk 7]
```diff
@@ -129,9 +124,7 @@ module.exports = (projectPersistence, biomeService) => ({
   getImpactedBiomes: async (projectId) => {
     const pId = parseInt(projectId, 10);
     if (!pId) {
-      const error = new Error('Invalid project id');
-      error.code = 400;
-      throw error;
+      throw new RestifyErrors.NotFoundError('Invalid project id');
```
Comment (erikasv) [inline services/backend/src/service/project.js:127]: bad reqeuest

File: services/backend/src/service/project_strategy.js [hunk 8]
```diff
@@ -10,9 +12,7 @@ module.exports = (strategyPersistence, restAPI) => ({
   createStrategy: async (projectId, strategy) => {
     const pId = parseInt(projectId, 10);
     if (!pId) {
-      const error = new Error('Invalid project id');
-      error.code = 400;
-      throw error;
+      throw new RestifyErrors.NotFoundError('Invalid project id');
```
Comment (erikasv) [inline services/backend/src/service/project_strategy.js:15]: bad request

File: services/backend/src/service/project_strategy.js [hunk 9]
```diff
@@ -38,9 +38,7 @@ module.exports = (strategyPersistence, restAPI) => ({
   getSelectedStrategiesGeoJson: async (projectId) => {
     const pId = parseInt(projectId, 10);
     if (!pId) {
-      const error = new Error('Invalid project id');
-      error.code = 400;
-      throw error;
+      throw new RestifyErrors.NotFoundError('Invalid project id');
```
Comment (erikasv) [inline services/backend/src/service/project_strategy.js:41]: bad request

**Review by erikasv — CHANGES_REQUESTED** [review]
persistence/project needs change for an error

File: services/backend/src/persistence/models/util/events.js [hunk 10]
```diff
@@ -1,13 +1,15 @@
...
  * This function is supposed to be usedn with this event:
  * http://bookshelfjs.org/index.html#Model-event-saving
  */
 const saving = (requiredFields, modelFields) => {
   const missing = requiredFields.filter((field) => !modelFields[field]);
   if (missing.length > 0) {
-    const error = new Error(`The following properties are missing a value: ${missing}`);
-    error.code = 400;
-    throw error;
+    throw new RestifyErrors.NotFoundError(
```
Comment (erikasv) [inline services/backend/src/persistence/models/util/events.js:10]: bad request

**Review by erikasv — CHANGES_REQUESTED** [review]

File: services/backend/src/persistence/models/util/events.js (hunk 10 above)
Comment (cazapatamar) [reply services/backend/src/persistence/models/util/events.js]: Fixed!

File: services/backend/src/service/project.js (hunk 6 above)
Comment (cazapatamar) [reply services/backend/src/service/project.js]: Fixed!

**Review by ManuelStardust — APPROVED** [review]
update works fine

**Review by erikasv — APPROVED** [review]

**Pull request merged**